src/ekat/mpi/ekat_comm.hpp

```diff
@@ -1,6 +1,10 @@
 #ifndef EKAT_COMM_HPP
 #define EKAT_COMM_HPP
 
+#include <ekat/ekat_config.h>
+
+#ifdef EKAT_ENABLE_MPI
```

Comment:
You are already compiling this  only conditionally, so what's the point of this? There are downstream files that include ekat_comm, so I think the cleanest solution is to provide a "serial" implementation of Comm, if MPI is not enabled. Either that, or we need to make the usage of Comm conditional to EKAT_ENABLE_MPI. See, for instance, `ekat_logger.hpp`.

Reply:
In Trilinos, they have a base class `Comm`, and concrete derived classes called `SerialComm` and `MpiComm`. Do you think we should do the same?

Reply:
I do agree with you that a "serial" implementation of the basic elements of MPI is the tried-and-true way to go. I was just trying to see what I could get away with in the EAGLES environment, where we don't invest in MPI communicators (because we're only computing on a single node in standalone mode).

I think we could certainly do what Trilinos does. Another idea is to put all of Comm's behavior in a `.cpp` file and have the build system select the right one.

Reply:
Maybe. There are a few places with `MPI_XYZ` in the hpp though. I think all the implementation section at the bottom of the hpp can be moved in the cpp file (since we use ETI anyways). You will still need an ifdef in the hpp when the type `MPI_Comm` is mentioned, and also we would have to declare MPI_Op and MPI_Datatype enums in the serial case, though.

I'm not sure what the best path is. Perhaps you could change all fcns that use Comm in ekat, and roll something like
```
#ifdef EKAT_ENABLE_MPI
void foo (const Comm& comm);
#else
void foo ();
#endif
```
(or even completely not define `foo` in the else branch, depending what it is). But I would remove the ifdefs from this file, since they are inconsistent with the approach of not compiling `ekat_comm.cpp` if no MPI is enabled.

Reply:
Yeah, I think you're right, actually. It seems to me that the best approach is for people who don't need MPI not to refer to anything that touches MPI. I've removed the guards from `ekat_comm.hpp` and added guards to files within EKAT that refer to it. I've verified that EKAT builds in a serial Haero configuration. So maybe this is all we need?